base/broadcast.jl

```diff
@@ -837,6 +837,9 @@ function __dot__(x::Expr)
     dotargs = map(__dot__, x.args)
     if x.head == :call && dottable(x.args[1])
         Expr(:., dotargs[1], Expr(:tuple, dotargs[2:end]...))
+    elseif x.head == :comparison
+        Expr(:comparison, (iseven(i) && dottable(arg) && arg isa Symbol ? Symbol('.', arg) : arg
```

Comment:
Also check isoperator(arg)

Reply:
Don't we already know we're in infix position here?

Reply:
With metaprogramming anything might happen. Welcome to my world... 😀